test/view/linear_distribute.cpp

```diff
@@ -0,0 +1,82 @@
+// Range v3 library
+//
+//  Gonzalo Brito Gadeschi 2017
+//
+//  Use, modification and distribution is subject to the
+//  Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at
+//  http://www.boost.org/LICENSE_1_0.txt)
+//
+// Project home: https://github.com/ericniebler/range-v3
+#include <cmath>
+#include <range/v3/algorithm/equal.hpp>
+#include <range/v3/view/linear_distribute.hpp>
+#include <range/v3/view/transform.hpp>
+#include <range/v3/view/iota.hpp>
+#include "../simple_test.hpp"
+#include "../test_utils.hpp"
+
+int main()
+{
+    using ranges::view::linear_distribute;
+
+    auto float_eq = [](double a, double b) {
+        return (int)std::floor(std::abs(a - b)) == 0;
+    };
+
+    {
+        auto irng = linear_distribute(0, 1, 2);
+        CHECK(ranges::size(irng) == std::size_t{2});
+        CONCEPT_ASSERT(ranges::ForwardRange<decltype(irng)>());
+        CONCEPT_ASSERT(ranges::SizedRange<decltype(irng)>());
+        auto il = {0, 1};
+        check_equal(irng, il);
+    }
+    {
+        auto irng = linear_distribute(1, 3, 3);
+        auto il = {1, 2, 3};
+        check_equal(irng, il);
+        CHECK(ranges::size(irng) == std::size_t{3});
+    }
+    {
+        auto irng = linear_distribute(0, 21, 22);
+        auto frng = linear_distribute(0., 21., 22);
+        check_equal(irng, frng | ranges::view::transform([](double v) { return int(v); }));
```

Comment:
`[](double v) { return int(v); }` could be `ranges::convert_to<int>{}` from `functional.hpp`.